contrib/hashicorp/consul/consul.go

```diff
@@ -38,18 +44,31 @@ func NewClient(config *consul.Config, opts ...ClientOption) (*Client, error) {
 	if err != nil {
 		return nil, err
 	}
-	return WrapClient(c, opts...), nil
+	return WrapClientWithConfig(c, config, opts...), nil
 }
 
 // WrapClient wraps a given consul.Client with a tracer under the given service name.
+// Deprecated: Replaced with WrapClientWithConfig such that config information can be used in span tags.
 func WrapClient(c *consul.Client, opts ...ClientOption) *Client {
 	cfg := new(clientConfig)
 	defaults(cfg)
 	for _, fn := range opts {
 		fn(cfg)
 	}
 	log.Debug("contrib/hashicorp/consul: Wrapping Client: %#v", cfg)
-	return &Client{c, cfg, context.Background()}
+	return &Client{c, nil, cfg, context.Background()}
+}
+
+// WrapClientWithConfig wraps a given consul.Client with a tracer under the given service name plus additional config info
+func WrapClientWithConfig(c *consul.Client, config *consul.Config, opts ...ClientOption) *Client {
```

Comment:
I don't think we should require a `*consul.Config` option as an argument here. We have no idea whether a customer has access to the original config here, and requiring it sort of defeats the purpose of wrapping an existing client. It makes much more sense to do this in `NewClient` than in `WrapClient`.

If we want to allow users to specify a config or an address, we can add that as another `ClientOption`.

Reply:
Looping in @ajgajg1134 because originally, we just wanted to deprecate the old wrapper function to include the config. This is a fair point that users may not have access to the config, so should we add the config as another option? We would need to make retroactive changes to the Kafka integrations found in #1979 because the same issue would apply there.